runhouse/resources/hardware/launcher_utils.py

```diff
@@ -434,6 +432,23 @@ def keep_warm(cls, cluster, mins: int):
             set_cluster_autostop_cmd = _cluster_set_autostop_command(mins)
             cluster.run_bash_over_ssh([set_cluster_autostop_cmd], node=cluster.head_ip)
 
+    @classmethod
+    def load_creds(cls):
+        """Loads the SSH credentials resource required for the launcher."""
+        try:
+            secret = rh.provider_secret("sky")
+            return secret
+
+        except ValueError:
+            # Create a new key pair required by Sky
+            path, _ = generate_ssh_keys()
+            secret = rh.provider_secret(
+                provider="sky", path=path, name=SSH_SKY_SECRET_NAME
+            )
+            secret.save()
```

Comment:
I think `secret.save()` saves the resource but you need `.write()` to write it to file system